skyplane/cli/cli_impl/init.py

```diff
@@ -80,21 +79,14 @@ def make_role_cmds(principal_id, subscription_id):
         if force_init:
             typer.secho("    Azure credentials will be re-initialized", fg="red", err=True)
             clear_azure_config(config, verbose=False)
-        if (
-            config.azure_enabled
-            and config.azure_subscription_id
-            and config.azure_tenant_id
-            and config.azure_client_id
-            and config.azure_client_secret
-        ):
+        if config.azure_enabled and config.azure_subscription_id and config.azure_principal_id and config.azure_client_id:
             typer.secho("    Azure credentials already configured! To reconfigure Azure, run `skyplane init --reinit-azure`.", fg="blue")
             return config
 
         # load credentials from environment variables or input
         defaults = {
-            "tenant_id": os.environ.get("AZURE_TENANT_ID") or config.azure_tenant_id,
             "client_id": os.environ.get("AZURE_CLIENT_ID") or config.azure_client_id,
-            "client_secret": os.environ.get("AZURE_CLIENT_SECRET") or config.azure_client_secret,
+            "principal_id": os.environ.get("AZURE_PRINCIPAL_ID") or config.azure_principal_id,
```

Comment:
Why do we read this from the environment variable? We should not ask for the principal_id at all and just generate it using `az identity` every time, both for interactive and non-interactive modes.